File: LeetCode/LeetCode 367. Valid Perfect Square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        low: int = 1
        high: int = num
        
        while low <= high:
            mid: int = int((low + high) >> 1)
            
            square: int = mid * mid
            
            if square == num:
                return True
            elif square > num:
                high = mid - 1
            else:
                low = mid + 1
        
        return False
            

File: LeetCode/LeetCode 367. Valid Perfect Square.js
/**
 * @param {number} num
 * @return {boolean}
 */
 var isPerfectSquare = function(num) {
    let first = 0.0, last = num;
    for(let step = 1; step <= 64; step++){
        let mid = (first + last) / 2;
        
        let square = Math.floor(mid) * Math.floor(mid);
        
        if(square === num){
            return true;
        } else if(square > num){
            last = mid;
        } else {
            first = mid;
        }
    }
    
    return false;
};